doc/source/tune/tutorials/tune-trial-checkpoint.rst

```diff
@@ -0,0 +1,197 @@
+To create an AIR checkpoint, one can either use :meth:`~ray.air.checkpoint.Checkpoint.from_dict`
+(Checkpoint a dictionary) or :meth:`~ray.air.checkpoint.Checkpoint.from_directory` APIs
+(Checkpoint a directory).
+
+.. TODO(ml-team): https://github.com/ray-project/ray/issues/33251
+.. warning:: When using ``from_directory``, the content of the checkpoint will be copied and moved
+    to a tune managed folder (<trial_name>/checkpoint_<epoch>). This may cause some inefficiency when
+    checkpoint is synced to driver node or the cloud. We are planning to work on it to address the
+    issue.
+
+.. tabbed:: Checkpoint a dictionary
+
+    .. literalinclude:: /tune/doc_code/trial_checkpoint.py
+        :language: python
+        :start-after: __function_api_checkpointing_start__
+        :end-before: __function_api_checkpointing_end__
+
+.. tabbed:: Checkpoint a directory
+
+    .. literalinclude:: /tune/doc_code/trial_checkpoint.py
+        :language: python
+        :start-after: __function_api_checkpointing_from_dir_start__
+        :end-before: __function_api_checkpointing_from_dir_end__
+
+In the above code snippet:
+
+- We implement *checkpoint saving* with :meth:`session.report(..., checkpoint=checkpoint) <ray.air.session.report>`. Note that every checkpoint must be reported alongside a set of metrics -- this way, checkpoints can be ordered with respect to a specified metric.
+- The saved checkpoint during training iteration `step` is saved to the path ``<local_dir>/<exp_name>/<trial_name>/checkpoint_<step>`` on the node on which training happens and can be further synced to a consolidated storage location depending on the :ref:`storage configuration <tune-storage-options>`.
+- We implement *checkpoint loading* with :meth:`session.get_checkpoint() <ray.air.session.get_checkpoint>`. This will be populated with a trial's latest checkpoint whenever Tune restores a trial. This happens when (1) a trial is configured to retry after encountering a failure, (2) the experiment is being restored, and (3) the trial is being resumed after a pause (ex: :doc:`PBT </tune/examples/pbt_guide>`).
+
+  .. TODO: for (1), link to tune fault tolerance guide. For (2), link to tune restore guide.
+
+.. note::
+    ``checkpoint_frequency`` and ``checkpoint_at_end`` will not work with Function API checkpointing.
+    These are configured manually with Function Trainable. For example, if you want to checkpoint every three
+    epochs, you can do so through:
+
+    .. literalinclude:: /tune/doc_code/trial_checkpoint.py
+        :language: python
+        :start-after: __function_api_checkpointing_periodic_start__
+        :end-before: __function_api_checkpointing_periodic_end__
+
+
+See :ref:`here for more information on creating checkpoints <air-checkpoint-ref>`.
+If using framework-specific trainers from Ray AIR, see :ref:`here <air-trainer-ref>` for
+references to framework-specific checkpoints such as `TensorflowCheckpoint`.
```

Comment:
Let's make this a `.. seealso::` and swap the location with the warning above.

swap with this warning: `When using ``from_directory``, the content of the checkpoint...`